aea/crypto/ethereum.py

```diff
@@ -94,10 +92,41 @@ def sign_transaction(self, message: str) -> bytes:
         :return: Signed message in bytes
         """
         m_message = encode_defunct(text=message)
-        signature = self.account.sign_message(m_message)
+        signature = self._account.sign_message(m_message)
         return signature
 
     def _generate_private_key(self) -> Account:
         """Generate a key pair for ethereum network."""
         account = Account.create()
         return account
+
+    @property
+    def token_balance(self) -> float:
```

Comment:
for now crypto objects bundle both the entity and the ledger access. later we can refactor and separate it into two